db/commitlog/commitlog.cc

```diff
@@ -608,6 +608,12 @@ auto db::commitlog::segment_manager::named_file::remove_file() noexcept {
     });
 }
 
+template<typename T>
+static void write(db::commitlog::output& out, T value) {
+    auto v = net::hton(value);
+    out.write(reinterpret_cast<const char*>(&v), sizeof(v));
```

Comment:
Please measure the impact using scylla perf-simple-query --mode write and look at instructions per op. If it grows too much, we'll need to double compile the whole thing, once for commits that fit in a sector and can use char* iterators, and once for the expensive iterators. Let's define the threshold as 200 instructions/op.

Reply:
Note the method above is just for header writes. All serialization goes through the normal fragmented output, just with a block-splitting iterator in the middle. Will measure, but it should not add that much. Note that I added the split-iterator just so we don't add a bunch of extra objects and stuff.

Reply:
Well, probably not definitive, since my machine is not exactly 100% bench reliable, but I get pretty equivalent results comparing master and this PR (after rebase). Perhaps a little more spread in this PR, though median remains pretty solid.

Reply:
@elcallio try setting the cpupower governor to `userspace`. It should set the cpu frequency to the lowest supported value and so reduce the jitter to minimum (no frequency auto-tuning due to load and/or temperature).